File: src/algorithms/maximum-achievable.js
/**
 * An integer x is called achievable if it can become equal to the parameter 
 * num after applying the following operation no more than t times:
 *   • Increase or decrease x by 1, and simultaneously increase or decrease
 *     num by 1.
 * Returns the maximum possible achievable number.
 * 
 * @param {number} num - A starting number
 * @param {number} t - A number of operations to perform
 * @returns {number} - Returns the maximum possible achievable number
 */
function theMaximumAchievableX (num, t) {
  return t * 2 + num;
}

File: src/algorithms/maximum-achievable.py
def the_maximum_achievable_x(num, t):
    """Given two integers, num and t. A number is achievable if it can become 
    equal to num after applying the following operation:
    • Increase or decrease the number by 1, and simultaneously increase or 
    decrease num by 1.
    Return the maximum achievable number after applying the operation at most t 
    times.

    Args:
        num (int): A positive integer.
        t (int): A positive integer.

    Returns:
        int: Returns the maximum achievable number.
    """
    return t * 2 + num
